BodoSQL/bodosql/tests/test_types/test_s3_tables_iceberg_catalog.py

```diff
@@ -0,0 +1,111 @@
+import pandas as pd
+
+import bodo
+import bodosql
+from bodo.tests.utils import (
+    assert_tables_equal,
+    check_func,
+    gen_unique_table_id,
+    pytest_s3_tables,
+    run_rank0,
+)
+from bodosql.bodosql_types.s3_tables_catalog import S3TablesConnectionType
+
+pytestmark = pytest_s3_tables
+
+
+def test_basic_read(memory_leak_check, s3_tables_catalog):
+    """
+    Test reading an entire Iceberg table from S3 Tables in SQL
+    """
+    bc = bodosql.BodoSQLContext(catalog=s3_tables_catalog)
+
+    def impl(bc, query):
+        return bc.sql(query)
+
+    py_out = pd.DataFrame(
+        {
+            "A": ["ally", "bob", "cassie", "david", None],
+            "B": [10.5, -124.0, 11.11, 456.2, -8e2],
+            "C": [True, None, False, None, None],
+        }
+    )
+
+    query = 'SELECT * FROM "read_namespace"."bodo_iceberg_read_test"'
```

Comment:
It would be good to list a docstring for what must exist and where at the top. In case we ever do an account cleanup/deletion. Also if there are creation steps including those would be great.